### PR TITLE
Added plugin for capacity dropdown storage

### PR DESCRIPTION
**Pull request by cloudbehl**
Signed-off-by: Ankush Behl <cloudbehl@gmail.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 1]
```diff
@@ -42,22 +40,37 @@ const QueriesMatchingCapacityView: QueryMapType = {
...
-    CAPACITY_USAGE_QUERIES[StorageDashboardQuery.STORAGE_CEPH_CAPACITY_PODS_QUERY],
-    CAPACITY_USAGE_QUERIES[StorageDashboardQuery.STORAGE_CEPH_CAPACITY_VMS_QUERY],
-  ],
 };
 
-export const CapacityCard: React.FC<DashboardItemProps> = ({
+const getItems = (plugin: any, flags: FlagsObject) =>
+  plugin.filter((e) => flags[e.properties.required]);
+
+const getCapacityQueries = (flags) => {
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]: Add types for flag which I think will be `FlagsObject` only

Comment (cloudbehl) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]: ack

File: frontend/packages/kubevirt-plugin/src/plugin.tsx [hunk 2]
```diff
@@ -191,6 +193,19 @@ const plugin: Plugin<ConsumedExtensions> = [
       required: FLAG_KUBEVIRT,
     },
   },
+  {
+    type: 'Dashboards/Storage/Capacity/Dropdown/Item',
+    properties: {
+      metric: 'VMs vs Pods',
+      queries: {
+        'VMs vs Pods': [
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/plugin.tsx]: why is this object and not array ? seems like the key is necessary as it is the same as `metric` ?

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 3]
```diff
@@ -42,22 +41,36 @@ const QueriesMatchingCapacityView: QueryMapType = {
...
     CAPACITY_USAGE_QUERIES[StorageDashboardQuery.STORAGE_CEPH_CAPACITY_USED_QUERY],
   ],
-  [CapacityViewType.CAPACITY_VMS_VS_PODS]: [
-    CAPACITY_USAGE_QUERIES[StorageDashboardQuery.STORAGE_CEPH_CAPACITY_PODS_QUERY],
-    CAPACITY_USAGE_QUERIES[StorageDashboardQuery.STORAGE_CEPH_CAPACITY_VMS_QUERY],
-  ],
 };
 
-export const CapacityCard: React.FC<DashboardItemProps> = ({
+const getItems = (plugin: DashboardStorageExtension[], flags: FlagsObject) =>
```
Comment (vojtechszocs) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]: Same as in #2476, nit: please use more sensible parameter names.

Will be fixed in follow-up.

**Comment by vojtechszocs** [comment]
There will be merge conflict with #2476 on added plugin SDK / typings for Ceph, please rebase.

**Comment by rawagner** [comment]
/hold
needs rebase and test is failing

**Comment by vojtechszocs** [comment]
@cloudbehl Please rebase this PR on top of #2476.

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 4]
```diff
@@ -10,6 +13,7 @@ import {
   DashboardItemProps,
   withDashboardResources,
 } from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { DashboardsStorageCapacityDropdownItem } from '@console/plugin-sdk';
```
Comment (vojtechszocs) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]: Note that this changes with #2592.

Plugin-specific extension types should live inside the corresponding plugin pkg.

**Comment by vojtechszocs** [comment]
The follow up to #2476 and this PR is #2592.

I'll update it once both 2476 and 2478 are merged.

**Pull request merged**